tests/social-share-button.test.js

```
@@ -0,0 +1,568 @@
+
+import { describe, it, expect, beforeEach, afterEach, vi } from "vitest";
+
+// We need to set up the DOM environment before importing the module
+// because the IIFE at the bottom of social-share-button.js runs on import.
+let SocialShareButton;
+
+beforeEach(async () => {
+  // Reset module registry so each test gets a fresh class with clean statics
+  vi.resetModules();
+  const mod = await import("../src/social-share-button.js");
+  SocialShareButton = mod.default || window.SocialShareButton;
```

ESLint

[error] 'window' is not defined.

(no-undef)